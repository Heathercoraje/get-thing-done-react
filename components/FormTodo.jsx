import React from 'react';
import styled from 'styled-components';

const FormTodo = ({ add, selected }) => {
  const onClickInput = (event) => {
    event.preventDefault();
    const changeEvent = event;
    add(event, selected);
    changeEvent.target[0].value = '';
  };

  return (
    <form onSubmit={onClickInput}>
      <input required placeholder="Enter your task" />
      <input type="submit" value="+" className="button-add" />
    </form>
  );
};

export default FormTodo;
