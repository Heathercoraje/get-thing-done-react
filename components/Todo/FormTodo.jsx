import React from 'react';
import styled from 'styled-components';
import HandleClickInput from '../../lib/HandleClickInput';

const Wrapper = styled.div`
	margin-top: auto;
`;

const FormTodo = ({ add }) => {
	const onClickInput = evt => {
		const inputTodo = document.getElementById('input-todo');
		evt.preventDefault();
		HandleClickInput(evt, add, inputTodo);
	};
	return (
		<Wrapper>
			<form onSubmit={onClickInput}>
				<input
					className="input-gen input-todo"
					id="input-todo"
					style={{ display: 'none' }}
					placeholder="Enter a new task"
				/>
				<input type="submit" value="+" className="button-add" />
			</form>
		</Wrapper>
	);
};

export default FormTodo;
