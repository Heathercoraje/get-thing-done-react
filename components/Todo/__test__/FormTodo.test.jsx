import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormTodo from '../FormTodo';

test('FormTodo should render correctly', () => {
	const component = shallow(<FormTodo />);
	const tree = toJson(component);
	expect(tree).toMatchSnapshot();
});

test('Add Button should have display:none property at default', () =>{
  const component = shallow(<FormTodo />);
  expect(component.find('input').first().prop('style')).toEqual({"display":"none"});
});
