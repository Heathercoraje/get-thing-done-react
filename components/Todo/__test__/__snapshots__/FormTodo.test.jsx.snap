// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTodo should render correctly 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="input-gen input-todo"
      id="input-todo"
      placeholder="Enter a new task"
      style={
        Object {
          "display": "none",
        }
      }
    />
    <input
      className="button-add"
      type="submit"
      value="+"
    />
  </form>
</styled.div>
`;
